File: src/domain/exceptions/oas_parser_exception.py
class OasParserException(Exception):
    INVALID_OAS_FORMAT = 1
    UNSUPPORTED_OAS_VERSION = 2
    FILE_NOT_FOUND = 3
    INVALID_FILE_CONTENT = 4

    def __init__(self, message: str = "OAS Parser error", code: int = 0, error_code: int = 0):
        super().__init__(message, code)
        self.error_code = error_code

    def get_error_code(self) -> int:
        return self.error_code

File: src/php/OasParserException.php
<?php

namespace TuEmpresa\LaravelAiApiIntegration\Domain\Exceptions;

class OasParserException extends \Exception {
    public const INVALID_OAS_FORMAT = 1;
    public const UNSUPPORTED_OAS_VERSION = 2;
    public const FILE_NOT_FOUND = 3;
    public const INVALID_FILE_CONTENT = 4;

    private int $errorCode;

    public function __construct(string $message = "OAS Parser error", int $code = 0, \Throwable $previous = null, int $errorCode = 0) {
        parent::__construct($message, $code, $previous);
        $this->errorCode = $errorCode;
    }

    public function getErrorCode(): int {
        return $this->errorCode;
    }
}
